com.unity.ml-agents/Runtime/Inference/BarracudaModelExtensions.cs

```diff
@@ -0,0 +1,141 @@
+            {
+                return (int)model.GetTensorByName(TensorNames.IsContinuousControlDeprecated)[0] > 0;
+            }
+            else
+            {
+                return model.outputs.Contains(TensorNames.ContinuousActionOutput) &&
+                    (int)model.GetTensorByName(TensorNames.ContinuousActionOutputShape)[0] > 0;
+            }
+        }
+
+        /// <summary>
+        /// Continuous action output size of the model.
+        /// </summary>
+        /// <param name="model">
+        /// The Barracuda engine model for loading static parameters.
+        /// </param>
+        /// <returns>Size of continuous action output.</returns>
+        public static int ContinuousOutputSize(this Model model)
+        {
+            if (model.UseDeprecated())
+            {
+                return (int)model.GetTensorByName(TensorNames.IsContinuousControlDeprecated)[0] > 0 ?
+                    (int)model.GetTensorByName(TensorNames.ActionOutputDeprecated)[0] : 0;
+            }
+            else
+            {
+                return (int)model.GetTensorByName(TensorNames.ContinuousActionOutputShape)[0];
+            }
+        }
+
+        /// <summary>
+        /// Continuous action output tensor name of the model.
+        /// </summary>
+        /// <param name="model">
+        /// The Barracuda engine model for loading static parameters.
+        /// </param>
+        /// <returns>Tensor name of continuous action output.</returns>
+        public static string ContinuousOutputName(this Model model)
+        {
+            if (model.UseDeprecated())
+            {
+                return TensorNames.ActionOutputDeprecated;
+            }
+            else
+            {
+                return TensorNames.ContinuousActionOutput;
+            }
+        }
+
+        /// <summary>
+        /// Check if the model has discrete action outputs.
+        /// </summary>
+        /// <param name="model">
+        /// The Barracuda engine model for loading static parameters.
+        /// </param>
+        /// <returns>True if the model has discrete action outputs.</returns>
+        public static bool HasDiscreteOutputs(this Model model)
+        {
+            if (model.UseDeprecated())
+            {
+                return (int)model.GetTensorByName(TensorNames.IsContinuousControlDeprecated)[0] == 0;
+            }
+            else
+            {
+                return model.outputs.Contains(TensorNames.DiscreteActionOutput) &&
+                    (int)model.GetTensorByName(TensorNames.DiscreteActionOutputShape)[0] > 0;
+            }
+        }
+
+        /// <summary>
+        /// Discrete action output size of the model.
+        /// </summary>
+        /// <param name="model">
+        /// The Barracuda engine model for loading static parameters.
+        /// </param>
+        /// <returns>Size of discrete action output.</returns>
+        public static int DiscreteOutputSize(this Model model)
```

Comment:
Might want to make it clearer (either in comments or method name) that this is the sum of the branch sizes, not the number of branches. That was confusing when its return value was assigned to `modelSumDiscreteBranchSizes` later.